### ISSUE TITLE
Random behavior of Error class and forEach loop

### ISSUE THREAD

**Issue by superexroid** [oven-sh/bun#3533]
### What version of Bun is running?

0.6.13

### What platform is your computer?

Linux 4.4.0-18362-Microsoft x86_64 x86_64

### What steps can reproduce the bug?

1. make a forEach loop
2. call a function inside loop
3. throw two identical error inside function

In 0.8% of attempts, these two identical errors have different stack traces!

```js
let counter = 0;
function foo(){
    let e1 = (new Error).stack.split("\n")[2];
    let e2 = (new Error).stack.split("\n")[2];
    if (e1 != e2){ // impossible ?! lets see!
        counter++;
    }
}
let range = [];
for (let i = 0; i < 100000; i++) {
    range.push(i);
}
range.forEach(i => {
    foo();
});
console.log(`${counter} out of ${range.length} attempts have this bug!`);
```

### What is the expected behavior?

above script should return
```
0 out of 100000 attempts have this bug!
```

### What do you see instead?

```
795 out of 100000 attempts have this bug!
```

### Additional information

other loops that works fine:
- for ... in
- every
- while

**Comment by Jarred-Sumner** [comment]
super interesting

I think it's a sourcemap bug.

**Comment by Jarred-Sumner** [comment]
I can confirm it doesn't happen in `jsc` shell.

<img width="296" alt="image" src="https://github.com/oven-sh/bun/assets/709451/a5058578-1723-4a65-8827-977bb1c496bf">

It seems to happen less when JIT is disabled
> 3064 out of 100000 attempts have this bug!
> 1214 out of 100000 attempts have this bug!

**Comment by superexroid** [comment]
> super interesting
> 
> I think it's a sourcemap bug.

yes it's introduced in ```bun-v0.6.12``` and probably #3441

**Comment by superexroid** [comment]
Fixed in ```bun-v1.0.0```
